corehq/apps/case_search/models.py

```diff
@@ -112,11 +118,17 @@ def _validate_multiple_terms(self):
     def _validate_daterange(self):
         if not self.is_daterange:
             return
-
         pattern = re.compile(r'__range__\d{4}-\d{2}-\d{2}__\d{4}-\d{2}-\d{2}')
-        match = pattern.match(self.value)
-        if not match:
-            raise CaseFilterError(_('Invalid date range format, {}'), self.key)
+        if self.has_multiple_terms:
+            for v in self.value:
+                if v:
+                    match = pattern.match(v)
+                    if not match:
+                        raise CaseFilterError(_('Invalid date range format, {}'), self.key)
+        else:
+            match = pattern.match(self.value)
+            if not match:
+                raise CaseFilterError(_('Invalid date range format, {}'), self.key)
```

Comment:
nit: you can dry this up a bit to prevent duplicating the pattern matching
```suggestion
        if self.has_multiple_terms:
            # don't validate empty values
            values = self._value_without_empty()
        else:
            values = [self.value]
        for v in values:
            match = pattern.match(v)
            if not match:
                raise CaseFilterError(_('Invalid date range format, {}'), self.key)
```